From: Fabien Tassin <fta@ubuntu.com>
Date: Fri, 25 Feb 2011 10:41:53 +0100
Subject: Fix the Webkit version in about:version

The build system expects the svn or git directories to be available at
build time.

Patch from Ubuntu, thanks to Fabien Tassin
---
 src/webkit/build/webkit_version.py |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/src/webkit/build/webkit_version.py b/src/webkit/build/webkit_version.py
index 7fdf9b32..b8c1af57 100755
--- a/src/webkit/build/webkit_version.py
+++ b/src/webkit/build/webkit_version.py
@@ -60,6 +60,15 @@ def GetWebKitRevision(webkit_dir, version_file):
       default_lastchange=None,
       directory=os.path.join(webkit_dir, version_file_dir),
       directory_regex_prior_to_src_url='webkit')
+  if version_info.url is None or version_info.url == 'unknown':
+    # parse the DEPS file
+    data = open("../../DEPS", "rb").read()
+    x = re.search(r'"webkit_revision": "(\d+)",', data)
+    rev = x.group(1) if x else "n-a"
+    x = re.search(r'"webkit_trunk": "(.*?)",', data)
+    branch = os.path.basename(x.group(1)) if x else "n-a"
+    return "%s@%s" % (branch, rev)
+
 
   if version_info.url == None:
     version_info.url = 'Unknown URL'
